### ISSUE TITLE
npm WARN @angular/flex-layout@2.0.0-rc.1 requires a peer of @angular/common@^2.2.3 but none was installed

### ISSUE THREAD

**Issue by alexjlockwood** [angular/flex-layout#344]
When I try to install the library I get the following warnings:

```
npm WARN @angular/flex-layout@2.0.0-rc.1 requires a peer of @angular/common@^2.2.3 but none was installed.
npm WARN @angular/flex-layout@2.0.0-rc.1 requires a peer of @angular/core@^2.2.3 but none was installed.
npm WARN @angular/flex-layout@2.0.0-rc.1 requires a peer of zone.js@^0.7.2 but none was installed.
```

**Comment by jotatoledo** [comment]
The rc.1 version was renamed, I think thats beta,7. If you want to use angular 4, then you need to use beta.8 of flex layout

**Comment by ThomasBurleson** [comment]
Please clean-install @angular/flex-layout v2.0.0-beta.8.

**Comment by gimox** [comment]
+1 very bad! clean it from npm

**Comment by ThomasBurleson** [comment]
@gimox - the Rc.1 was deprecated and we attempted to clean it from the registry. 

*  How are you installing the Rc.1 version? 
*  How do you propose we improve the cleaning of that invalid release from the npm registry?

**Comment by gimox** [comment]
How are you installing the Rc.1 version?
- new project with angular cli
- npm install @angular/flex-layout --save
- npm update (now it add rc-1)

How do you propose we improve the cleaning of that invalid release from the npm registry?
 
bump to a upper version rc-2
or
 unpublish it  https://docs.npmjs.com/cli/unpublish

**Comment by nicksloan** [comment]
If the code is not ready to be in RC, then you need to just acknowledge that the 2.0.0 release is ruined and scrap it. The best thing to do would be to release 2.0.1-beta1 or whatever the first release of the next version number would be. You should never go down again, and unpublishing seems like a bad idea.

**Comment by gimox** [comment]
rc1 is the same as beta6, it's a publish error, it can safe unpublish

**Comment by nicksloan** [comment]
@gimox Except for the fact that some people will certainly have rc1 in their package.json files, so you'll be breaking their code on unpublish.

**Comment by ThomasBurleson** [comment]
Beta6 is an old outdated release. Devs should absolute nstall nightly build
for now; see wiki for details.

**Comment by un33k** [comment]
You need to watch out to ensure you are not indirectly installing the outdated RC.1.

flex-layout "2.0.0-rc.1" is chronologically greater than any "2.0.0-beta.X".

Therefore, if you have something like "@angular/flex-layout": "^2.0.0-beta.9" in your package.json, then, yarn/npm install will automatically look at the "^" mark and bring down the latest minor release.

It seems both **beta** and **rc** releases are considered minor. So, the latest chronological minor release on npm would be "2.0.0-rc.1 and it will get installed instead as per "^".

**Note that "2.0.0-rc.1" is the same as "2.0.0-beta.5".** (you would be using very very old code)

**Workaround:**
Pin down your "@angular/flex-layout": "^2.0.0-beta.9" by removing the "^" mark.


TL;DR
Also, verify the final install using (ng --version)

**TEST:** 
1. Ensure you have this in your package.json ("@angular/flex-layout": "^2.0.0-beta.9")
2. yarn/npm install
3. ng --version 
@angular/cli: 1.4.4
node: 6.11.0
os: darwin x64
@angular/animations: 4.4.4
@angular/cdk: 2.0.0-beta.11
@angular/common: 4.4.4
@angular/compiler: 4.4.4
@angular/core: 4.4.4
**@angular/flex-layout: 2.0.0-rc.1** (remove the "^" mark from package.json and try agian)
@angular/forms: 4.4.4
@angular/http: 4.4.4
@angular/material: 2.0.0-beta.11
@angular/platform-browser: 4.4.4
@angular/platform-browser-dynamic: 4.4.4
@angular/router: 4.4.4
@angular/service-worker: 1.0.0-beta.16
@angular/cli: 1.4.4
@angular/compiler-cli: 4.4.4
@angular/language-service: 4.4.4

**Comment by ThomasBurleson** [comment]
Do not install beta 9 if you need Angular 5 compatibility. Those fixes are
in nightly builds and will be released in upcoming beta.10.

**Comment by nicksloan** [comment]
@ThomasBurleson The concern is not new users, but all of your existing users who you are forcing to come here to figure out how to fix the problem that you created by breaking the semver contract. Many of them have already upgraded to rc.1 erroneously, and the problem won't realistically be fixed until you release rc.2 or, as I have suggested, release 2.0.1-beta.1, which is the only correct course of action.

This is a very disappointing way to manage this project.

**Comment by un33k** [comment]
@nicksloan This is still a beta project for a reason. It is not ready yet & mistakes can happen.  I am glad that you stumbled on this thread and figured out your way.

@ThomasBurleson I for sure appreciate all the time you put in this project and looking forward to seeing the RC.2 release.

Base on this (+ve) note, let's conclude this thread. Pls.

**Comment by ThomasBurleson** [comment]
The currently nightly release is stable with Angular 5 and will be an important release for the community. Unfortunately, my team does not control the final testing and actual release processes for this library. 

> Due to the last 2-3 weeks of intense efforts on Angular Material, Flex-Layout release is pending. We hope to see the next release next week.

I do acknowledge that we need to remove the Beta.1 tag OR publish the next release as 2.0.1-beta.1.
And please note that my team wants @angular/Flex-layout to be a wonderful, versatile Layout tool for angular developers.

> We have some amazing additions coming in Q1, 2018.
